src/cargo/ops/cargo_update.rs

```diff
@@ -41,6 +45,162 @@ pub fn generate_lockfile(ws: &Workspace<'_>) -> CargoResult<()> {
     Ok(())
 }
 
+pub fn update_manifests(
+    ws: &mut Workspace<'_>,
+    opts: &UpdateOptions<'_>,
+) -> CargoResult<HashMap<String, Version>> {
+    let mut upgrades = HashMap::new();
+
+    if !opts.incompatible {
+        return Ok(upgrades);
+    }
+
+    // Updates often require a lot of modifications to the registry, so ensure
+    // that we're synchronized against other Cargos.
+    let _lock = ws
+        .gctx()
+        .acquire_package_cache_lock(CacheLockMode::DownloadExclusive)?;
+
+    let mut registry = PackageRegistry::new(opts.gctx)?;
+    registry.lock_patches();
+
+    for member in ws.members_mut() {
+        debug!("updating manifest for {}", member.name());
+
+        let new_summary = member
+            .manifest()
+            .summary()
+            .clone()
+            .map_dependencies(|dependency| {
+                let req = if let OptVersionReq::Req(current) = dependency.version_req() {
+                    let query = crate::core::dependency::Dependency::parse(
+                        dependency.package_name(),
+                        None,
+                        dependency.source_id().clone(),
+                    )
+                    .unwrap();
+
+                    let possibilities = {
+                        loop {
+                            match registry.query_vec(&query, QueryKind::Exact) {
+                                std::task::Poll::Ready(res) => {
+                                    break res.unwrap();
+                                }
+                                std::task::Poll::Pending => registry.block_until_ready().unwrap(),
+                            }
+                        }
+                    };
+
+                    let latest = if !possibilities.is_empty() {
+                        possibilities
+                            .iter()
+                            .map(|s| s.as_summary())
+                            .map(|s| s.version().clone())
+                            .max()
+                    } else {
+                        None
+                    };
+
+                    if let Some(latest) = latest.clone() {
+                        if !current.matches(&latest) {
```

Comment:
I'm not seeing this restricted to only semver operator

Reply:
Changed now. At the moment I am _only_ allowing the caret operator. Not sure if that is too restrictive.

Reply:
Yes, `^` is the "semver" operator and is the only one we should be supporting atm (hence the name change of the flag)